hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/container/ListSubcommand.java

```diff
@@ -76,6 +76,11 @@ public class ListSubcommand extends ScmSubcommand {
       description = "Container replication (ONE, THREE for Ratis, " +
           "rs-6-3-1024k for EC)")
   private String replication;
+  
+  @Option(names = {"--json"},
+      description = "Output the entire list in JSON array format",
```

Comment:
The original is still json (and can still be streamed into jq), it just doesn't represent a valid json file if it is saved that way. In that sense I think the flag here is confusing. I would just make json list the default and only behavior. We did something similar in HDDS-5824 and did not consider it a breaking change, since the original was a "bug". With 2.0 coming up such a change makes sense as well.

Reply:
Existing acceptance tests would need to be updated to account for this.

Reply:
Yeah, I was trying to avoid changing the existing behavior of the CLI, as it would be good to tread CLI output as a contract. I ran into JSON parsing errors when analyzing a bunch of returned containers. `jq` works fine with streaming in the container JSONs, but it would be good to have JSON array as per speck as well.